src/model/entities/device.ts

```diff
@@ -1,9 +1,92 @@
 import { StatusCode } from '../enums/status-code'
 
-export interface Device {
-  id: string
-  name: string
-  isConnected: boolean
-  statusCode: StatusCode
-  statusMessage: string
+export abstract class Device {
+  public _id: string
+  private _name: string
+  private _isConnected: boolean
+  private _statusCode: StatusCode
+  private _statusMessage: string
+
+  constructor(id: string, name: string, isConnected: boolean, statusCode: StatusCode, statusMessage: string) {
+    this._id = id
+    this._name = name
+    this._isConnected = isConnected
+    this._statusCode = statusCode
+    this._statusMessage = statusMessage
+  }
+
+  public abstract connect(): void
+  
+  public get id(): string {
+    return this._id
+  }
```

Comment:
There is nothing wrong by using setters and getters like this, but I want to point out that in our other entity classes, we use methods, e.g. `getId()` and `setId()`. It would be nice to have it being consistent. That way, we also avoid the `_` prefix for the internal attributes.

Reply:
Refactoring to methods!